### ISSUE TITLE
Custom types converted incorrectly into OpenAPI 3

### ISSUE THREAD

**Issue by reieRMeister** [kminami/apib2swagger#67]
# Nominal

Rendering an API Blueprint document to OpenAPI3 with flag `--open-api-3` converts a custom type like this:

```
## MyType (object)
+ id: `4711` (required, string)
+ someProperty (required, nullable, object)
    + someList (required, array)
        + (object)
            + id: `some example`(required, string)
```

into this:

```
    MyType:
      type: object
      required:
        - id
        - someProperty
      properties:
        id:
          type: string
          example: 4711
        someProperty:
          type: object
          nullable: true
          required:
            - someList
          properties:
            someList:
              type: array
              items:
                type: object
                required: 
                  - id
                properties:
                  id:
                    type: string
                    example: some example
```

# Actual

Rendering an API Blueprint document to OpenAPI3 with flag `--open-api-3` converts a custom type like this:

```
## MyType (object)
+ id: `4711` (required, string)
+ someProperty (required, nullable, object)
    + someList (required, array)
        + (object)
            + id: `some example`(required, string)
```

results in:

```
    MyType:
      type: object
      required:
        - id
        - someProperty
      properties:
        id:
          type: string
        someProperty:
          type: object
          required:
            - someList
          properties:
            someList:
              type: array
              items: {}
          nullable: true
```

Please note the missing definition in line 16 and the missing example values.

**Comment by kminami** [comment]
Thank you for reporting!

I have implemented example conversion at v1.15.0.

For the missing definition, we should use `fixed-type` attribute.
e.g. 

```Apia
## MyType (object)
+ id: `4711` (required, string)
+ someProperty (required, nullable, object)
    + someList (required, array, fixed-type)
        + (object)
            + id: `some example`(required, string)
```

results in:

```yaml
    MyType:
      type: object
      required:
        - id
        - someProperty
      properties:
        id:
          type: string
          example: '4711'
        someProperty:
          type: object
          required:
            - someList
          properties:
            someList:
              type: array
              items:
                type: object
                required:
                  - id
                properties:
                  id:
                    type: string
                    example: some example
          nullable: true
```

**Comment by reieRMeister** [comment]
Looks good to me!
